### PR TITLE
Adding support to bind multiple listeners to backfill run events

### PR DESCRIPTION
**Pull request by AndrewZeitler**
Adding binding for multiple listeners to backfill events. This will allow us to add custom slack messages and actions based on backfill events

### CODE REVIEW

**Comment by AndrewZeitler** [comment]
> What is driving this change? Do you want to add a new listener?
> 
> Also curious at which company you are using this?

I am at Faire. We have a need to send a message directly out to those on our team when they run a backfill to also create one in another environment. Right now there doesn't seem to be another way to do this. Was talking with @shellderp about this approach

File: service/src/main/kotlin/app/cash/backfila/dashboard/BackfillStateToggler.kt [hunk 1]
```diff
@@ -59,9 +60,13 @@ class BackfillStateToggler @Inject constructor(
     }
 
     if (desiredState == RUNNING) {
-      slackHelper.runStarted(Id(id), caller.principal)
+      for (listener in backfillRunListeners) {
```
[the remark this answers is not in the document]
Comment (AndrewZeitler) [reply service/src/main/kotlin/app/cash/backfila/dashboard/BackfillStateToggler.kt:63]: Sounds good. Was following Faire's style guide which does not like forEach 😅

File: service-self-backfill/src/test/kotlin/app/cash/backfila/service/selfbackfill/SelfBackfillTestingModule.kt [hunk 2]
```diff
@@ -47,6 +49,9 @@ class SelfBackfillTestingModule : KAbstractModule() {
       web_url_root = "",
       slack = null,
     )
+    newMultibinder<BackfillRunListener>()
+            .addBinding()
```
Comment (shellderp) [inline service-self-backfill/src/test/kotlin/app/cash/backfila/service/selfbackfill/SelfBackfillTestingModule.kt:53]: your indents seem too big everywhere

File: service/src/main/kotlin/app/cash/backfila/dashboard/BackfillStateToggler.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (shellderp) [reply service/src/main/kotlin/app/cash/backfila/dashboard/BackfillStateToggler.kt]: yea long story..

**Review by shellderp — APPROVED** [review]
lgtm with indent fix

File: service/src/main/kotlin/app/cash/backfila/dashboard/BackfillStateToggler.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpawliszyn) [reply service/src/main/kotlin/app/cash/backfila/dashboard/BackfillStateToggler.kt]: Well now you have to tell me LOL

**Pull request merged**